CDP4JsonSerializer/Cdp4JsonSerializer.cs

```diff
@@ -248,43 +282,48 @@ public T Deserialize<T>(Stream contentStream)
-            T data;
-            using (var streamReader = new StreamReader(contentStream))
-            using (var jsonTextReader = new JsonTextReader(streamReader))
-            {
-                var sw = new Stopwatch();
-                sw.Start();
-                data = serializer.Deserialize<T>(jsonTextReader);
-                Logger.Trace("Deserialize from stream in {0} [ms]", sw.ElapsedMilliseconds);
-            }
-
+            var sw = new Stopwatch();
+            sw.Start();
+            var data = JsonSerializer.Deserialize<T>(contentStream, this.JsonSerializerOptions);
+            Logger.Trace("Deserialize from stream in {0} [ms]", sw.ElapsedMilliseconds);
             return data;
         }
 
         /// <summary>
-        /// Create a <see cref="JsonSerializer"/>
+        /// Convenience method that deserializes the passed in JSON content stream
         /// </summary>
+        /// <typeparam name="T">
+        /// The type info for which deserialization will be performed
+        /// </typeparam>
+        /// <param name="contentString">
+        /// The content string.
+        /// </param>
         /// <returns>
-        /// an instance of <see cref="JsonSerializer"/>
+        /// The the deserialized instance of the specified Type
         /// </returns>
-        protected JsonSerializer CreateJsonSerializer()
+        public T Deserialize<T>(string contentString)
         {
-            Logger.Trace("initializing JsonSerializer");
-            var serializer = new JsonSerializer
+            if (this.RequestDataModelVersion == null || this.MetaInfoProvider == null)
             {
-                ContractResolver = new CamelCasePropertyNamesContractResolver(),
-                NullValueHandling = NullValueHandling.Ignore
-            };
-            
-            Logger.Trace("register converters");
-            serializer.Converters.Add(new ThingSerializer(this.MetaInfoProvider, this.RequestDataModelVersion));
-            serializer.Converters.Add(new ClasslessDtoSerializer(this.MetaInfoProvider, this.RequestDataModelVersion));
-            serializer.Converters.Add(new ClassKindConverter());
-            serializer.Converters.Add(new StringEnumConverter());
-
-            return serializer;
+                throw new InvalidOperationException("The supported version or the metainfo provider has not been set. Call the Initialize method to set them.");
+            }
+
+            var sw = new Stopwatch();
+            sw.Start();
+            var data = JsonSerializer.Deserialize<T>(contentString, this.JsonSerializerOptions);
+            Logger.Trace("Deserialize from stream in {0} [ms]", sw.ElapsedMilliseconds);
+            return data;
+        }
+
+        /// <summary>
+        /// Initialize the <see cref="JsonSerializerOptions" /> property
+        /// </summary>
+        public virtual void InitializeJsonSerializerOptions()
+        {
+            this.JsonSerializerOptions = SerializerOptions.Copy();

```

Comment:
I don't see the value of having the `SerializerOptions` class. Without having looked at the server code, webapp or ime code, i don;t think we need this. 

I also see that this method is `virtual`, giving the impression that implementers can or should override this. I doubt we need or want that.

Reply:
Done.